sdcm/provision/scylla_yaml/auxiliaries.py

```diff
@@ -0,0 +1,155 @@
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.
+#
+# See LICENSE for more details.
+#
+# Copyright (c) 2021 ScyllaDB
+
+
+import logging
+from typing import Literal, List, Union, Optional
+
+import attr
+
+from sdcm.provision.common.builders import ParamsBuilder, builder_property
+from sdcm.sct_config import SCTConfiguration
+
+SEED_PROVIDERS = [
+    'org.apache.cassandra.locator.SimpleSeedProvider',
+    'org.apache.cassandra.locator.GossipingPropertyFileSnitch',
+    'org.apache.cassandra.locator.Ec2Snitch',
+    'org.apache.cassandra.locator.Ec2MultiRegionSnitch',
+    'org.apache.cassandra.locator.GoogleCloudSnitch',
+    'org.apache.cassandra.locator.RackInferringSnitch',
+]
+SASLAUTHD_AUTHENTICATOR = 'com.scylladb.auth.SaslauthdAuthenticator'
+LOGGER = logging.getLogger(__file__)
+
+
+@attr.attrs(kw_only=True, order=False, auto_attribs=True, init=True, eq=True, repr=True)
+class SeedProvider:  # pylint: disable=too-few-public-methods
+    class_name: Literal[
+        'org.apache.cassandra.locator.SimpleSeedProvider',
+        'org.apache.cassandra.locator.GossipingPropertyFileSnitch',
+        'org.apache.cassandra.locator.Ec2Snitch',
+        'org.apache.cassandra.locator.Ec2MultiRegionSnitch',
+        'org.apache.cassandra.locator.GoogleCloudSnitch',
+        'org.apache.cassandra.locator.RackInferringSnitch',
+    ]
+    parameters: List[dict] = None
+
+
+@attr.attrs(kw_only=True, order=False, auto_attribs=True, init=True, eq=True, repr=True)
+class ServerEncryptionOptions:  # pylint: disable=too-few-public-methods
+    internode_encryption: Literal['all', 'none', 'dc', 'rack'] = 'none'
+    certificate: str = 'conf/scylla.crt'
+    keyfile: str = 'conf/scylla.key'
+    truststore: str = None
+    priority_string: str = None
+    require_client_auth: bool = False
+
+
+@attr.attrs(kw_only=True, order=False, auto_attribs=True, init=True, eq=True, repr=True)
+class ClientEncryptionOptions:  # pylint: disable=too-few-public-methods
+    enabled: bool = False
+    certificate: str = 'conf/scylla.crt'
+    keyfile: str = 'conf/scylla.key'
+    truststore: str = None
+    priority_string: str = None
+    require_client_auth: bool = False
+
+
+@attr.attrs(kw_only=True, order=False, auto_attribs=True, init=True, eq=True, repr=True)
+class RequestSchedulerOptions:  # pylint: disable=too-few-public-methods
+    throttle_limit: int = None
+    default_weight: int = 1
+    weights: int = 1
+
+
+EndPointSnitchType = Literal[
+    'org.apache.cassandra.locator.SimpleSnitch',
+    'org.apache.cassandra.locator.GossipingPropertyFileSnitch',
+    'org.apache.cassandra.locator.PropertyFileSnitch',
+    'org.apache.cassandra.locator.Ec2Snitch',
+    'org.apache.cassandra.locator.Ec2MultiRegionSnitch',
+    'org.apache.cassandra.locator.RackInferringSnitch'
+]
+
+
+@attr.attrs(kw_only=True, order=False, str=False, init=True)
+class ScyllaYamlBuilderBase(ParamsBuilder):
```

Comment:
Can we rename all the "ScyllaYaml" stuff to the "ScyllaConf" or "ScyllaConfig"? I don't think it is good idea to stick to the end file format. We should stick to the idea of the file.